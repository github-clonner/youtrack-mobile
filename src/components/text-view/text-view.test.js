import React from 'react';

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import TextView from './text-view';

describe('<TextView/>', () => {

  let wrapper;
  let textMock;

  describe('Render', () => {
    beforeEach(() => {
      textMock = 'A'.repeat(100);
      wrapper = doShallow(textMock);
    });

    it('should match a snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render content with `Show more...`', () => {
      expect(findByTestId('textMoreContent')).toHaveLength(1);
      expect(findByTestId('textMoreShowMore')).toHaveLength(1);
    });

    it('should render content with `Show more...` if there is not more to show (take a threshold into account)', () => {
      textMock = 'A'.repeat(wrapper.DEFAULT_CUT_LENGTH + wrapper.THRESHOLD + 1);
      wrapper = doShallow(textMock);
      expect(findByTestId('textMoreContent')).toHaveLength(1);
      expect(findByTestId('textMoreShowMore')).toHaveLength(0);
    });

    it('should render content only', () => {
      textMock = 'A'.repeat(20);
      wrapper = doShallow(textMock);

      expect(findByTestId('textMoreContent')).toHaveLength(1);
      expect(findByTestId('textMoreShowMore')).toHaveLength(0);
    });

  });


  function findByTestId(testId) {
    return wrapper && wrapper.find({testID: testId});
  }

  function doShallow(text = '', maxLength = 80) {
    return shallow(
      <TextView
        maxLength={maxLength}
        text={text}
      />
    );
  }
});
