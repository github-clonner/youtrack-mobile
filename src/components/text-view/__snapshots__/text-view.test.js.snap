// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextView/> Render should match a snapshot 1`] = `
<Text
  testID="textMoreContent"
>
  <Text>
    AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA...
  </Text>
  <Text
    onPress={[Function]}
    style={
      Object {
        "color": "#0f5b99",
      }
    }
    testID="textMoreShowMore"
  >
      Show more…  
  </Text>
</Text>
`;
